File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Commands/Querying/HowToQueryDatabase.cs
using System.Collections.Generic;

using Raven.Abstractions.Data;
using Raven.Client.Document;
using Raven.Json.Linq;

namespace Raven.Documentation.Samples.ClientApi.Commands.Querying
{
	public class HowToQueryDatabase
	{
		private interface IFoo
		{
			#region query_database_1
			QueryResult Query(
				string index,
				IndexQuery query,
				string[] includes = null, 
				bool metadataOnly = false,
				bool indexEntriesOnly = false);
			#endregion
		}

		public HowToQueryDatabase()
		{
			using (var store = new DocumentStore())
			{
				QueryResult result;

				#region query_database_2
				result = store.DatabaseCommands.Query("Orders/Totals", new IndexQuery
				{
					Query = "Company:companies/1"
				});

				List<RavenJObject> users = result.Results; // documents resulting from this query - orders
				#endregion

				#region query_database_3

				result = store.DatabaseCommands.Query("Orders/Totals", new IndexQuery(),
					new[]
					{
						"Company",
						"Employee"
					});

				List<RavenJObject> referencedDocs = result.Includes; // included documents - companies and employees
				#endregion
			}
		} 
	}
}

File: Documentation/3.0/Samples/java/src/test/java/net/ravendb/clientapi/commands/querying/HowToQueryDatabase.java
package net.ravendb.clientapi.commands.querying;

import java.util.Collection;
import java.util.List;

import net.ravendb.abstractions.data.IndexQuery;
import net.ravendb.abstractions.data.QueryResult;
import net.ravendb.abstractions.json.linq.RavenJObject;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.document.DocumentStore;


public class HowToQueryDatabase {

  @SuppressWarnings("unused")
  private interface IFoo {

    //region query_database_1
    public QueryResult query(String index, IndexQuery query);

    public QueryResult query(String index, IndexQuery query, String[] includes);

    public QueryResult query(String index, IndexQuery query, String[] includes, boolean metadataOnly);

    public QueryResult query(String index, IndexQuery query, String[] includes, boolean metadataOnly, boolean indexEntriesOnly);
    //endregion
  }

  @SuppressWarnings("unused")
  public HowToQueryDatabase() throws Exception {
    try (IDocumentStore store = new DocumentStore()) {
      QueryResult result;
      //region query_database_2
      result = store.getDatabaseCommands().query("Orders/Totals", new IndexQuery("Company:companies/1"));

      List<RavenJObject> users = result.getResults(); // documents resulting from this query - orders
      //endregion

      //region query_database_3
      result = store.getDatabaseCommands().query("Orders/Totals", new IndexQuery(), new String[] { "Company", "Employee" });

      Collection<RavenJObject> referencedDocs = result.getIncludes(); // included documents - companies and employees
      //endregion
    }
  }
}
